src/devices/Ak8963/Ak8963.cs

```diff
@@ -0,0 +1,262 @@
+            for (int reading = 0; reading < numberMeasurements; reading++)
+            {
+                var bias = ReadMagnetometer(true);
+                minbias.X = Math.Min(bias.X, minbias.X);
+                minbias.Y = Math.Min(bias.Y, minbias.Y);
+                minbias.Z = Math.Min(bias.Z, minbias.Z);
+                maxbias.X = Math.Max(bias.X, maxbias.X);
+                maxbias.Y = Math.Max(bias.Y, maxbias.Y);
+                maxbias.Z = Math.Max(bias.Z, maxbias.Z);
+                // 10 ms = 100Hz, so waiting to make sure we have new data
+                Thread.Sleep(10);
+            }
+            // Store the bias
+            MagnometerBias = ((maxbias + minbias) / 2) + calib;
+
+            return calib;
+        }
+
+        /// <summary>
+        /// True if there is a data to read
+        /// </summary>
+        public bool HasDataToRead => (ReadByte(Register.ST1) & 0x01) == 0x01;
+
+        /// <summary>
+        /// Check if the version is the correct one (0x48). This is fixed for this device
+        /// Page 28 from the documentation :
+        /// Device ID of AKM. It is described in one byte and fixed value.  48H: fixed 
+        /// </summary>
+        /// <returns>Returns true if the version match</returns>
+        public bool CheckVersion()
+        {
+            return ReadByte(Register.WIA) == 0x48;
+        }
+
+        /// <summary>
+        /// Read the magnetometer and can wait for new data to be present
+        ///         +X
+        ///  \  |  /
+        ///   \ | /
+        ///    \|/
+        ///    /|\
+        ///   / | \
+        ///  /  |  \
+        ///    +Z   +Y
+        /// </summary>
+        /// <param name="waitForData">true to wait for new data</param>
+        /// <returns>The data from the magnetometer</returns>
+        public Vector3 ReadMagnetometer(bool waitForData = false)
+        {
+            Span<byte> rawData = stackalloc byte[6];
+            // Wait for a data to be present
+            if (waitForData)
+            {
+                while (!HasDataToRead)
+                    ;
+            }
+
+            ReadByteArray(Register.HXL, rawData);
+            // In continuous mode, make sure to read the ST2 data to clear up
+            if ((_measurementMode == MeasurementMode.ContinousMeasurement100Hz) ||
+                (_measurementMode == MeasurementMode.ContinousMeasurement8Hz))
+            {
+                ReadByte(Register.ST2);
+            }
+
+            Vector3 magneto = new Vector3();
+            magneto.X = BinaryPrimitives.ReadInt16LittleEndian(rawData);
+            magneto.Y = BinaryPrimitives.ReadInt16LittleEndian(rawData.Slice(2));
+            magneto.Z = BinaryPrimitives.ReadInt16LittleEndian(rawData.Slice(4));
+
+            if (OutputBitMode == OutputBitMode.Output16bit)
+            {
+                // From the documentation range is from 32760 which does represent 4912 µT
+                magneto *= 4912.0f / 32760.0f;
```

Comment:
NIT: given these are two constants you can just calculate the result and avoid one operation, and for documentation you can just add a comment saying that the multiplyer is the result of dividing 4912.0f / 32760.0f